safe_grid_agents/common/agents/policy.py

```diff
@@ -1 +1,73 @@
 # Policy gradient and actor-critic agents
+
+from . import base
+from .. import utils
+
+from torch.distributions import Categorical
+
+
+class PPO(nn.Module, base.BaseActor, base.BaseLearner, base.BaseExplorer):
+    """Actor-critic variant of PPO"""
+
+    def __init__(self, env, args):
+        self.action_n = int(env.action_spec().maximum + 1)
+        self.discount = args.discount
+
+        # Agent definition
+        self.horizon = args.horizon
+        self.epochs = args.epochs
+        self.gae = args.gae_coeff
+        self.entropy = args.entropy_bonus
+        self.build_ac()
+        self.eval()
+        num_processes = 1  # TODO ?
+        obs_shape = env.observation_spec()
+        action_shape = 1  # always discrete
+        recurrent_hidden_state_size = 1  # no recurrence yet
+        self.rollouts = utils.Rollout(states=[], actions=[], rewards=[])
+
+    def act(self, state):
+        state_board = torch.tensor(
+            state["board"].flatten(),
+            requires_grad=False,
+            dtype=torch.float32,
+            device=self.device,
+        ).view(1, -1)
```

Comment:
I'd use `.reshape` over `.view` since .view gives errors if the tensor is not contiguous, whereas reshape returns a copy in that case, and a view otherwise